presto-connector-tests/src/main/java/com/facebook/presto/connector/unittest/BaseMetadataTest.java

```diff
@@ -0,0 +1,161 @@
+    {
+        return new SchemaTableName(schemaName, tableName);
+    }
+
+    @Test
+    default void testEmptyMetadata()
+            throws Exception
+    {
+        ConnectorSession session = new TestingConnectorSession(ImmutableList.of());
+
+        withMetadata(
+                ImmutableList.of(
+                        metadata -> assertEquals(metadata.listSchemaNames(session), systemSchemas()),
+                        metadata -> assertEquals(metadata.listTables(session, null), ImmutableList.of())));
+    }
+
+    /*
+     * Arguably, this belongs in MetadataTableTest. Unfortunately many
+     * connectors don't support CREATE_TABLE, but do support CREATE_TABLE_AS.
+     * As a result, MetadataTableTest is written in terms of CREATE_TABLE_AS,
+     * and is annotated with @RequiredFeatures({..., CREATE_TABLE_AS, ...) at
+     * the class level.
+     *
+     * Moving this to MetadataTableTest would require either:
+     * 1. Annotating every other methos than this as requiring CREATE_TABLE_AS.
+     * 2. Adding a way to unrequire previously required dependencies.
+     *
+     * 1 is cumbersome, 2 adds confusing semantics and surface area for weird
+     * bugs. Putting the test here seems like a good-enough solution.
+     */
+    @Test
+    @RequiredFeatures({CREATE_TABLE, DROP_TABLE})
+    default void testCreateDropTable()
+            throws Exception
+    {
+        ConnectorSession session = new TestingConnectorSession(ImmutableList.of());
+        String tableName = "table";
+        SchemaTableName schemaTableName = tableInDefaultSchema(tableName);
+
+        ConnectorTableMetadata tableMetadata = new ConnectorTableMetadata(
+                schemaTableName,
+                ImmutableList.of(
+                        new ColumnMetadata("bigint_column", BIGINT),
+                        new ColumnMetadata("double_column", DOUBLE)),
+                getTableProperties());
+
+        withSchemas(session, ImmutableList.of(schemaTableName.getSchemaName()),
+                ImmutableList.of(
+                        metadata -> metadata.createTable(session, tableMetadata),
+                        metadata -> assertEquals(getOnlyElement(metadata.listTables(session, schemaTableName.getSchemaName())), schemaTableName),
+                        metadata -> metadata.dropTable(session, metadata.getTableHandle(session, schemaTableName))));
+    }
+
+    default SchemaTablePrefix prefixOfSchemaName(SchemaTableName schemaTableName)
+    {
+        return new SchemaTablePrefix(schemaTableName.getSchemaName());
+    }
+
+    default SchemaTablePrefix prefixOf(SchemaTableName schemaTableName)
+    {
+        return new SchemaTablePrefix(schemaTableName.getSchemaName(), schemaTableName.getTableName());
+    }
+
+    default void withMetadata(List<Consumer<ConnectorMetadata>> consumers)
+    {
+        consumers.forEach(this::withMetadata);
+    }
+
+    default List<String> schemaNamesOf(SchemaTableName... schemaTableNames)
```

Comment:
Why it is here and not in some utility class?

Reply:
It's not clear that it's needed outside of this class. It can be moved in the future.